src/mysql/connection.cr

```diff
@@ -1,6 +1,37 @@
 require "socket"
 
 class MySql::Connection < DB::Connection
+  class ConnectionError < Exception; end
+
+  class PacketError < ConnectionError
+    getter :packet
+
+    def initialize(packet : ReadPacket, message)
+      @packet = packet
+      super(message)
+    end
+  end
+
+  class UnexpectedPacketError < PacketError
+    getter :status
```

Comment:
I would stick with `getter status : UInt8` style

Reply:
I'm admittedly inexperience with Crystal so I'll take the style advice, see 7fadad9d680345cd0db4c6f553f37d890ca273f1.